python/ccxt/base/decimal_to_precision.py

```diff
@@ -33,18 +33,24 @@
 PAD_WITH_ZERO = 6
 
 
-def decimal_to_precision(n, rounding_mode=ROUND, precision=None, counting_mode=DECIMAL_PLACES, padding_mode=NO_PADDING):
-    assert precision is not None
+def decimal_to_precision(n, rounding_mode=ROUND, numPrecisionDigits=None, counting_mode=DECIMAL_PLACES, padding_mode=NO_PADDING):
+    assert numPrecisionDigits is not None, 'numPrecisionDigits should not be None'
+
+    if isinstance(numPrecisionDigits, str):
+        numPrecisionDigits = float(numPrecisionDigits)
+    assert isinstance(numPrecisionDigits, float) or isinstance(numPrecisionDigits, decimal.Decimal) or isinstance(numPrecisionDigits, numbers.Integral), 'numPrecisionDigits has an invalid number'
+
     if counting_mode == TICK_SIZE:
-        assert(isinstance(precision, float) or isinstance(precision, decimal.Decimal) or isinstance(precision, numbers.Integral) or isinstance(precision, str))
+        assert numPrecisionDigits > 0, 'negative or zero numPrecisionDigits can not be used with TICK_SIZE precisionMode'
     else:
-        assert(isinstance(precision, numbers.Integral))
+        assert isinstance(numPrecisionDigits, numbers.Integral)
+
     assert rounding_mode in [TRUNCATE, ROUND]
     assert counting_mode in [DECIMAL_PLACES, SIGNIFICANT_DIGITS, TICK_SIZE]
     assert padding_mode in [NO_PADDING, PAD_WITH_ZERO]
+    # end of checks
 
-    if isinstance(precision, str):
-        precision = float(precision)
+    precision = numPrecisionDigits  # "precision" variable name was in signature, but to make function signature similar to php/js, I had to change the argument name to "numPrecisionDigits". however, the below codes use "precision" variable name, so we have to assign that name here (you can change the usage of 'precision' variable name below everywhere, but i've refrained to do that to avoid many changes)
```

Comment:
@carlosmiei @ttodua from a usage perspective in python, changing the function signature is potentially a breaking change depending on usage (if used with kwargs), failing with `TypeError: decimal_to_precision() got an unexpected keyword argument 'precision'`  (guess how i know :laughing: )

example (assuming proper imports
``` python
decimal_to_precision(
                amount,
                rounding_mode=TRUNCATE,
                precision=5,
                counting_mode=DECIMAL_PLACES,
            )
```

nothing that can't be worked around easily in user land - but something that maybe should reflect in versioning (making this a non-minor update?)

Reply:
yes, we had different signatures across langs for this method and python had different name of arg, so I've renamed it to be like other langs. however, I am not aware of the heavy dependency on internal methods by users, so I might have missed that nuance, thanks for the report. I'll re-evaluate to rename it back if we consider that there is some significant users reliance on that.

Reply:
@carlosmiei lmk how you decide, i can rename it back, just i wanted to have unified argument names for all langs, because it's the way we should have things, but still, depends on what do you think.

Reply:
@xmatthias it will be reverted to `precision` here shortly: https://github.com/ccxt/ccxt/pull/26547